src/components/Explorer/queryTemplate.js

```diff
@@ -54,86 +110,99 @@ JOIN leagues using(leagueid)
       'player_matches.hero_id',
       'player_matches.account_id',
       'leagues.name leaguename',
-    ].filter(Boolean).join(',\n')}
+    ].filter(Boolean).join(',\n') : ''}
 FROM matches
 JOIN match_patch using(match_id)
 JOIN leagues using(leagueid)
 JOIN player_matches using(match_id)
 JOIN heroes on heroes.id = player_matches.hero_id
 LEFT JOIN notable_players ON notable_players.account_id = player_matches.account_id AND notable_players.locked_until = (SELECT MAX(locked_until) FROM notable_players)
 LEFT JOIN teams using(team_id)
-${organization || (group && group.key === 'organization') ? 'JOIN team_match ON matches.match_id = team_match.match_id AND (player_matches.player_slot < 128) = team_match.radiant JOIN teams teams2 ON team_match.team_id = teams2.team_id' : ''}
-${(select && select.join) ? select.join : ''}
-${(select && select.joinFn) ? select.joinFn(props) : ''}
+${organization || (groupArray !== null && groupArray.length > 0 && groupArray[0] && groupArray.some(x => x.key === 'organization')) ?
+    'JOIN team_match ON matches.match_id = team_match.match_id AND (player_matches.player_slot < 128) = team_match.radiant JOIN teams teams2 ON team_match.team_id = teams2.team_id' : ''}
+${validate(selectArray) ? selectArray.map(x => (x.joinFn ? x.joinFn(props) : '')).join('') : ''}
+${validate(selectArray) ? selectArray.map(x => (x.join ? x.join : '')).join('') : ''}
 WHERE TRUE
-${select ? `AND ${select.value} IS NOT NULL` : ''}
-${minPatch ? `AND match_patch.patch >= '${minPatch.value}'` : ''}
-${maxPatch ? `AND match_patch.patch <= '${maxPatch.value}'` : ''}
-${hero ? `AND player_matches.hero_id = ${hero.value}` : ''}
-${player ? `AND player_matches.account_id = ${player.value}` : ''}
-${league ? `AND matches.leagueid = ${league.value}` : ''}
-${playerPurchased ? `AND (player_matches.purchase->>'${playerPurchased.value}')::int > 0` : ''}
-${minDuration ? `AND matches.duration >= ${minDuration.value}` : ''}
-${maxDuration ? `AND matches.duration <= ${maxDuration.value}` : ''}
-${side ? `AND (player_matches.player_slot < 128) = ${side.value}` : ''}
-${result ? `AND ((player_matches.player_slot < 128) = matches.radiant_win) = ${result.value}` : ''}
-${team ? `AND notable_players.team_id = ${team.value}` : ''}
-${organization ? `AND team_match.team_id = ${organization.value} AND (player_matches.player_slot < 128) = team_match.radiant` : ''}
-${laneRole ? `AND player_matches.lane_role = ${laneRole.value}` : ''}
-${region ? `AND matches.cluster IN (${region.value.join(',')})` : ''}
-${minDate ? `AND matches.start_time >= extract(epoch from timestamp '${new Date(minDate.value).toISOString()}')` : ''}
-${maxDate ? `AND matches.start_time <= extract(epoch from timestamp '${new Date(maxDate.value).toISOString()}')` : ''}
-${tier ? `AND leagues.tier = '${tier.value}'` : ''}
+${validate(selectArray) ? selectArray.map(x => `AND ${x.value} IS NOT NULL `).join('') : ''}
+${minPatch ? templ`match_patch.patch >= '${minPatch}'` : ''}
```

Comment:
perhaps you can allow the templating function to return an empty string if e.g. minPatch is falsy?  Then you could avoid this outer ternary.